### PR TITLE
Fix analysis

### PR DESCRIPTION
**Pull request by qle2**
Hey @Dlux804 I have combined the two functions called multipredict and replicate_model in analysis into one function so we don't have to run them 10 times. I also added a function that makes importance graph.

Please let me know what you think of them. Once you approve, I will make unittest for them.

### CODE REVIEW

File: core/analysis.py [hunk 1]
```diff
@@ -29,59 +29,124 @@ def predict(regressor, train_features, test_features, train_target, test_target)
...
+            # Axis labels and title
+            plt.ylabel('Importance');
+            plt.xlabel('Variable');
+            # plt.title('EXP:' + exp + '  Variable Importances');
+
+            # ax = plt.axes()
+            ax.xaxis.grid(False)  # remove just xaxis grid
+
+            plt.savefig(model_name + '-importance.png')
+            return plt
```
Comment (Dlux804) [inline core/analysis.py]: Please also have it return the variable importance in some sort of text format.  This way we can store it as an attribute of the ML instance and export later.  Data frame is the preferred format (```varimp```)

File: core/analysis.py [hunk 2]
```diff
@@ -29,59 +29,124 @@ def predict(regressor, train_features, test_features, train_target, test_target)
...
+        else:
+            pass
+    else:
+        pass
+
+def replicate_multi(regressor, train_features, test_features, train_target, test_target, n=5):
+    """
+    Objective: Run model n times. Return dictionary of r2, mse and rmse average and standard deviation and predict each
+    point multiple times to calculate uncertainty. I know, not my proudest copy and paste.
```
Comment (Dlux804) [inline core/analysis.py]: Please remove "I know, not my proudest..."

File: core/analysis.py [hunk 3]
```diff
@@ -29,59 +29,124 @@ def predict(regressor, train_features, test_features, train_target, test_target)
...
+
+    :param regressor: Regression model
+    :param n: Number of times the model is run
+    :return:
```
Comment (Dlux804) [inline core/analysis.py]: Please add what it returns and in what format it is returned, i.e data frame, .png, etc.

File: core/analysis.py [hunk 4]
```diff
@@ -29,59 +29,124 @@ def predict(regressor, train_features, test_features, train_target, test_target)
...
+        'rmse_avg': rmse.mean(),
+        'rmse_std': rmse.std(),
+        'time_avg': t.mean(),
+        'time_std': t.std()
+    }
+    print('Average R^2 = %.3f' % stats['r2_avg'], '+- %.3f' % stats['r2_std'])
+    print('Average RMSE = %.3f' % stats['rmse_avg'], '+- %.3f' % stats['rmse_std'])
+    print()
 
+    return stats, pva_multi, fit_time
```
Comment (Dlux804) [inline core/analysis.py]: Why are you returning ```fit_time```?  That is the variable that contains the fit time of the last run only.  Please instead return the array ```t``` that contains the raw fit time values for all runs.

File: core/models.py [hunk 5]
```diff
@@ -51,9 +51,9 @@ def run(self, tune=False):
             # jobs = int(input('Input the number of processing cores to use. (-1) to use all.'))
 
             # FIXME Unfortunate hard code deep in the program.
-            folds = 10
-            iters = 100
-            jobs = 30  # for bayes, max jobs = folds.
+            folds = 2
+            iters = 3
+            jobs = -1  # for bayes, max jobs = folds.
```
Comment (Dlux804) [inline core/models.py]: I assume you changed these just for testing purposes?

File: core/models.py [hunk 6]
```diff
@@ -121,6 +123,7 @@ def store(self):
 
         # save graphs
         self.graphM.savefig(name+'PvAM')
+        # self.impgraph.savefig(name+'impgraph')
```
Comment (Dlux804) [inline core/models.py]: Why is this commented out? We want the graph.

File: fragments/sim_frag.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+You know, big boys big girls stuff
+"""
+# Read csv
+all_df = pd.read_csv('short_freesolv.csv')
+# print(all_df)
+# Make molob using PandasTools
+df = PandasTools.AddMoleculeColumnToFrame(all_df, 'smiles', 'molobj', includeFingerprints=True)
+print(all_df)
+molob = all_df['molobj']
```
Comment (Dlux804) [inline fragments/sim_frag.py]: This work seems incomplete.  I suggest removing the folder fragments/ from this merge request and creating a separate branch for the fragment feature.

**Comment by qle2** [comment]
@Dlux804 Please review once more.

File: core/models.py [hunk 8]
```diff
@@ -74,16 +74,17 @@ def run(self, tune=False):
 
 
         #Variable importance for rf and gdb
-        # analysis.impgraph(self.regressor, train_features, train_target, self.feature_list)
+        self.impgraph = analysis.impgraph(self.algorithm, self.regressor, train_features, train_target, self.feature_list)
```
Comment (Dlux804) [inline core/models.py]: @qle2 ```analysis.impgraph()``` now returns the graph and the raw data for importance.  Please store these separately.  Right now they both get stored (likely as a tuple) to ```self.impgraph``` 

Similarly, please update ```model.store()``` to write the ```varimp``` to a csv.

**Comment by Dlux804** [comment]
This Closes #1

**Pull request merged**